File: testing/syntax-error/java/syntax_error.java
/*
Raspberry Pi Foundation
Developed to be used alongside Isaac Computer Science,
part of the National Centre for Computing Education
Usage licensed under CC BY-SA 4

Note: This file is designed to be copied out and compiled on your machine.
In order for the program to compile properly in some IDEs, you need to ensure that the
class filename is the same as the class name with the main method in it.

To run this file you need to:
1. Copy the contents
2. Create a Java project in the IDE of your choice (Eclipse, for example)
3. Create a new Java Class that uses the same name as the class containing the main method in this program
4. Paste the contents into the new Java Class
5. Save and run the program
*/

import java.util.Scanner;

class SyntaxErrors
{
    // The main method is the entry point for all Java programs
    public static void main(String[] args)
    {
        // Use Scanner for reading user input from the console
        Scanner scanner = new Scanner(System.in);

        // Demonstrates a syntax error
        System.out.printl("Please enter your name: ");
        String name = scanner.nextLine();
        if ("Ada".equals(name)) {
            System.out.println("Hi Ada, you are one of my heroes!");
        }
        else {
            System.out.println("Hello " + name);
        }
    }


}

File: testing/syntax-error/visual-basic/syntax_error.vb
' Raspberry Pi Foundation
' Usage licensed under CC BY-NC-SA 4.0
' Note: This file is designed to be copied out and compiled on your machine.
' To run this file you need to:
' 1. Copy the contents
' 2. Paste them into the Visual Basic IDE of your choice (Visual Studio, for example)
' 3. Compile the program
' 4. Run the program

Module Program

    ' The Main subroutine is the default entry point for a VB program
    Sub Main()
        ' Demonstrates a syntax error
        Console.WriteLine("Please enter your name: ")
        Dim name As String = Console.ReadLine()

        If name = "Ada" Then
            Console.WriteLine("Hi Ada, you are one of my heroes!")
        Else
            Console.WriteLine("Hello " & name)
        End If

    End Sub


End Module
